File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOWatchDialInfoModel.kt
//
//  IDOWatchDialInfoModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///

// MARK: - IDOWatchDialInfoModel

open class IDOWatchDialInfoModel(
    blockSize: Int,
    familyName: String,
    format: Int,
    height: Int,
    sizex100: Int,
    width: Int
) : IDOBaseModel {

    /// Compression block size
    @SerializedName("block_size")
    var blockSize: Int = blockSize

    /// Family name (maximum 10 bytes)
    @SerializedName("family_name")
    var familyName: String = familyName

    /// Color format
    @SerializedName("format")
    var format: Int = format

    /// Screen height (pixel size)
    @SerializedName("height")
    var height: Int = height

    /// Size in 100x increments
    @SerializedName("sizex100")
    var sizex100: Int = sizex100

    /// Screen width (pixel size)
    @SerializedName("width")
    var width: Int = width

    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOWatchDialInfoModel.swift
//
//  IDOWatchDialInfoModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/19.
//

import Foundation

// MARK: - IDOWatchDialInfoModel
@objcMembers
public class IDOWatchDialInfoModel: NSObject, IDOBaseModel {
    /// Compression block size
    public var blockSize: Int
    /// Family name (maximum 10 bytes)
    public var familyName: String
    /// Color format
    public var format: Int
    /// Screen height (pixel size)
    public var height: Int
    /// Size in 100x increments
    public var sizex100: Int
    /// Screen width (pixel size)
    public var width: Int

    enum CodingKeys: String, CodingKey {
        case blockSize = "block_size"
        case familyName = "family_name"
        case format = "format"
        case height = "height"
        case sizex100 = "sizex100"
        case width = "width"
    }

    public init(blockSize: Int, familyName: String, format: Int, height: Int, sizex100: Int, width: Int) {
        self.blockSize = blockSize
        self.familyName = familyName
        self.format = format
        self.height = height
        self.sizex100 = sizex100
        self.width = width
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

